File: src/main/java/com/philhanna/flashcards/domain/Deck.java
package com.philhanna.flashcards.domain;

import java.util.List;

/**
 * A deck of flash cards having a title and a list of cards.
 */
public interface Deck {

   /**
    * Returns the deck title
    */
   public String getTitle();

   /**
    * Returns an ordered list of the cards in the deck
    */
   public List<Card> getCards();
}

File: src/flashcards/domain/deck.py
# flashcards.domain.deck

from abc import ABC, abstractmethod
from typing import Sequence

from flashcards.domain.card import Card


class Deck(ABC):

    @property
    @abstractmethod
    def title(self) -> str: ...

    @property
    @abstractmethod
    def cards(self) -> Sequence[Card]: ...
